File: easy/20-valid-parentheses/20-valid-parentheses.rb
# @param {String} s
# @return {Boolean}
def is_valid(s)
    brackets = {'{' => '}', '[' => ']', '(' => ')' }
    stack = []
    s.each_char do |bracket|
        if brackets.key? bracket
            stack << brackets[bracket]
        else
            return false if stack.pop != bracket
        end
    end
    stack.size == 0
end

File: easy/20-valid-parentheses/20-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
const isValid = function(s) {
   const brackets = {
       '(': ')',
       '{': '}',
       '[': ']'
   }
   const stack = [];
   for (const bracket of s) {
       if (brackets[bracket]) {
            stack.push(brackets[bracket]); 
       } else {
           if (stack.pop() !== bracket) return false;
       }
   }
    return !stack.length
};